src/components/SignupModal.jsx

```diff
@@ -0,0 +1,180 @@
+                  type="password"
+                  placeholder="Password"
+                  name="password"
+                  value={user.password}
+                  onChange={this.handleChange}
+                  className={
+                    error.password ? "error form-control" : "form-control"
+                  }
+                  error={error.password}
+                  required
+                />
+                <TextInput
+                  type="password"
+                  placeholder="Confirm password"
+                  name="confirm_password"
+                  value={user.confirm_password}
+                  onChange={this.handleChange}
+                  className={
+                    error.confirm_password
+                      ? "error form-control"
+                      : "form-control"
+                  }
+                  error={error.confirm_password}
+                  required
+                />
+
+                <SubmitButton
+                  label="Sign up"
+                  type="submit"
+                  fetching={fetching}
+                  className="btn m-auto btn-primary btn-block"
+                />
+              </form>
+              <p className="text-center">OR</p>
+              <p className="d-flex flex-column">
+                <button
+                  type="button"
+                  className="btn text-left m-1 ah-google-button btn-block"
+                >
+                  <i className="fab fa-google" /> &ensp; Sign up with Google
+                </button>
+                <button
+                  type="button"
+                  className="btn text-left m-1 ah-twitter-button btn-block"
+                >
+                  <i className="fab fa-twitter" /> &ensp; Sign up with Twitter
+                </button>
+                <button
+                  type="button"
+                  className="btn text-left m-1 ah-facebook-button btn-block"
+                >
+                  <i className="fab fa-facebook" /> &ensp; Sign up with Facebook
+                </button>
+                <br />
+                <small className="text-center">
+                  Already have an account?{" "}
+                  <a
+                    className="p-1"
+                    href="/"
+                    data-toggle="modal"
+                    data-dismiss="modal"
+                    data-target="#ahSignInModal"
+                  >
+                    Sign In
+                  </a>
+                </small>
+              </p>
+            </div>
+          )}
+        </div>
+      </div>
+    );
+  }
+}
+
+SignupModal.propTypes = {
+  dispatch: PropTypes.func.isRequired,
+  message: PropTypes.string,
+  error: PropTypes.objectOf(PropTypes.array),
+  success: PropTypes.bool,
+  fetching: PropTypes.bool
+};
+
+SignupModal.defaultProps = {
+  error: {},
+  message: null,
+  success: false,
+  fetching: false
+};
+
+const mapStateToProps = state => state.authSignup;
```

Comment:
Refactor `mapStateToProps` for easier understanding of what you are trying to access in this case. I can see in `rootReducer.js` you have renamed your reducer to `authSignup`. Make use of this `authSignup` to access the props you desire to use in this case.

const mapStateToProps = ({ authSignup }) => {
  const { message, ... } = authSignup;
  return {
    message,
    ...
  };
};

Reply:
I actually was struggling with this. Thank you very much. I will amend this as well